akka-remote/src/main/scala/akka/remote/artery/RemoteInstrument.scala

```diff
@@ -0,0 +1,204 @@
+/*
+ * Copyright (C) 2016 Lightbend Inc. <http://www.lightbend.com>
+ */
+
+package akka.remote.artery
+
+import java.{ util ⇒ jutil }
+
+import akka.actor.{ ActorRef, ExtendedActorSystem }
+import akka.util.{ ByteString, OptionVal }
+
+/**
+ * INTERNAL API
+ *
+ * Part of the monitoring SPI which allows attaching metadata to outbound remote messages,
+ * and reading in metadata from incoming messages.
+ *
+ * Multiple instruments are automatically handled, however they MUST NOT overlap in their idenfitiers.
+ */
+abstract class RemoteInstrument {
+  /**
+   * Instrument identifier.
+   *
+   * MUST be >=0 and <32.
+   *
+   * Values between 0 and 7 are reserved for Akka internal use.
+   */
+  def identifier: Byte
+
+  /**
+   * Can be used to instrument a place in the code to collect "tell-side" data,
+   * The returned value will be attached to the [[OutboundEnvelope]] under this instruments identifier,
+   * and passed into the `remoteMessageSent` method at a later point in time (and space).
+   *
+   * @return `context` that will be passed into [[remoteMessageSent]] once the message reaches the Encoder
+   */
+  def remoteActorTold(actorRef: ActorRef, message: Any, sender: ActorRef): OptionVal[AnyRef]
```

Comment:
Correct, it's not – see discussion on top-level of this ticket with Bjorn.
Effectively this WOULD be called in `send()` or somewhere way earlier than in coder, to collect some near-actor-metadata. We won't put this into this SPI though, monitoring guys can do it anyway. Will remove this method